odoo/addons/base/ir/ir_ui_view.py

```diff
@@ -237,6 +238,12 @@ def _compute_model_data_id(self):
             view = self.browse(data['res_id'])
             view.model_data_id = data['id']
 
+    def _search_model_data_id(self, operator, value):
+        name = 'name' if isinstance(value, basestring) else 'id'
+        domain = [('model', '=', 'ir.ui.view'), (name, operator, value)]
+        datas = self.env['ir.model.data'].search(domain)
+        return ('id', 'in', datas.mapped('res_id'))
```

Comment:
Isn't the return value supposed to be a domain, not just a leaf, so that you can return arbitrary domains when needed?

```python
return [('id', 'in', datas.mapped('res_id'))]
```

Is this roughly as fast as before when you search in the list of views (which searches on `model_data_id` by default)?

Finally, I guess the pun of using `datas` as a variable name is intended? :smiley_cat: